### ISSUE TITLE
versioning

### ISSUE THREAD

**Issue by fjenett** [nodebox/live#20]
Adding a way of versioning would allow to embed a specific version of a project into external websites but also keep working on it without breaking the published version.

**Comment by fjenett** [comment]
I just found the "releases" in the help files … but that seems disabled for me (can not see the icon).

**Comment by fdb** [comment]
You're right, once a project is released it can break easily when doing changes. There is some structure in place to provide versioned releases. However, we disabled the mechanism in the UI because it's too confusing.

The current idea is to allow users to create snapshots of a project. Each time they create a snapshot, a number is increased sequentially (like a build number). The website can then chose how to embed the project:

- Embed a numbered snapshot (e.g. snapshot 23)
- Embed the latest snapshot (this will update each time you press the "publish" button inside NBL)
- Embed the live version (only for development, since it can break easily)

**Comment by fjenett** [comment]
That would work for me. … but it's hidden/disabled atm, right?

**Comment by fdb** [comment]
Yes, the functionality is not exposed from the UI. Also, we currently have a much more "heavyweight" versioning process instead of the snapshotting functionality that I proposed. So it will take a bit of time to rewrite it and expose the UI.
